operator/pkg/values/translation/translation.go

```diff
@@ -138,40 +138,88 @@ func GetGlobalMonitoring(m *platform.GlobalMonitoring) *ValuesBuilder {
 	return &globalMonitoring
 }
 
-// SetScannerAnalyzerValues sets values in "sv" based on "analyzer".
-func SetScannerAnalyzerValues(sv *ValuesBuilder, analyzer *platform.ScannerAnalyzerComponent) {
-	if analyzer.GetScaling() != nil {
-		scaling := analyzer.GetScaling()
-		sv.SetInt32("replicas", scaling.Replicas)
-
-		autoscaling := NewValuesBuilder()
-		if scaling.AutoScaling != nil {
-			switch *scaling.AutoScaling {
-			case platform.ScannerAutoScalingDisabled:
-				autoscaling.SetBoolValue("disable", true)
-			case platform.ScannerAutoScalingEnabled:
-				autoscaling.SetBoolValue("disable", false)
-			default:
-				autoscaling.SetError(fmt.Errorf("invalid spec.scanner.replicas.autoScaling %q", *scaling.AutoScaling))
-			}
+// SetScannerComponentDisabled sets the disabled values for scanner configurations
+func SetScannerComponentDisabledValue(sv *ValuesBuilder, scannerComponent *platform.ScannerComponentPolicy) {
+	if scannerComponent != nil {
+		switch *scannerComponent {
+		case platform.ScannerComponentDisabled:
+			sv.SetBoolValue("disable", true)
+		case platform.ScannerComponentEnabled:
+			sv.SetBoolValue("disable", false)
+		default:
+			sv.SetError(fmt.Errorf("invalid spec.scanner.scannerComponent %q", *scannerComponent))
```

Comment:
The path in this error message is not necessarily correct anymore, right?

Reply:
adjusted the error message.